### ISSUE TITLE
Error output from FritzboxConfig.py: 1 - 13: not found

### ISSUE THREAD

**Issue by Ldwg2002** [RenWal/fritzbox-munin-fast#5]
Zwei Router (im VPN) und zwei kleinere WiFi-Netze möchte ich gerne mit fritzbox-munin/fast überwachen. Die fritzbox-munin plugins laufen, nicht ganz problemlos mit 06.8x. Details hier https://github.com/Tafkas/fritzbox-munin/issues/62#issue-1530814006  Möglichkeiten und Ergebnisse stellen mich prinzipiell zufrieden. Aber die Stichworte "fast, multigraph" haben mich zu diesem fork gebracht, für

* einen Überblick über den **Verkehr am WAN/Router,** gerne Info über die **Qualität der DSL-Leitung**
* Info zur **Auslastung im "shared medium" WiFi** an allen Antennen (AccessPoints, bis zu 8 Stk). 

| | Diese drei plugins sollten genau das leisten: |
|-|-|
| _link_saturation.py | Multigraph plugin, showing saturation of WAN uplink and downlink by QoS priority |
| _traffic.py | Similar to fritzbox_link_saturation, but single-graph and without QoS monitoring. |
|_wifi_load.py | Multigraph plugin, showing for 2.4GHz and 5GHz <br /> / WiFi uplink and downlink bandwidth usage / neighbor APs on same and on different channels |

### config # Was mache ich falsch? ###

> **See the plugin files for plugin-specific configuration options.**  
Völlig übersehen hatte ich diesen Hinweis in der Anleitung !!!! Den Issue https://github.com/RenWal/fritzbox-munin-fast/blob/master/src/fritzbox_dsl.py#L10 habe ich gelesen, aber eine funktionsfähige Config bekomme ich nicht hin. 

```
vi /etc/munin/munin.conf
[munin.plugin;test00fritzfast]
     address 127.0.0.1      # address  localhost # take from plugin.conf
     use_node_name      no  # access by node-name=yes ip-address=no
```

```
vi /etc/munin/plugin-conf.d/munin-node
[test00fritzfast_*]  # env.host_name connects to entry in /etc/munin/munin.conf
env.host_name      test00fritzfast
env.fritzbox_ip    192.168.xyz.1   ### 3490os07.30 ### router
env.fritzbox_user       trusted
env.fritzbox_password   passwd
env.fritzbox_use_tls    false
env.dsl_modes      capacity   snr   damping   errors   crc 
env.ecostat_modes  cpu   temp   ram 
env.energy_modes   power   devices   uptime 
env.energy_product DSL
env.traffic_remove_max   1 
env.wifi_freqs     24 
env.wifi_modes   freqs    neighbors 
```

### Anmerkungen und Tests ###

**wifi_modes.py** kopiert wird **neighbors**, ist das ein Schreibfehler? **neighbours**  
**fritzfast_traffic.py** kann es sein, dass down und up vertauscht sind? https://github.com/Tafkas/fritzbox-munin/commit/d7929f15fc15d7e132288b262fc025d12a5f498a  

```
ls /usr/share/munin/plugins/fritzfast* 
/usr/share/munin/plugins/fritzfast_connection_uptime.py
/usr/share/munin/plugins/fritzfast_dsl.py
/usr/share/munin/plugins/fritzfast_ecostat.py
/usr/share/munin/plugins/fritzfast_energy.py
/usr/share/munin/plugins/fritzfast_link_saturation.py
/usr/share/munin/plugins/fritzfast_smart_home_temperature.py
/usr/share/munin/plugins/fritzfast_traffic.py
/usr/share/munin/plugins/fritzfast_wifi_load.py
 ls -l /usr/share/munin/plugins/fritzfast* 
-rwxr-xr-x 1 munin munin 2219 26. Jan 19:59 /usr/share/munin/plugins/fritzfast_connection_uptime.py
ls /usr/share/munin/plugins/test00fritzfast*
test00fritzfast_connection_uptime.py  test00fritzfast_link_saturation.py
test00fritzfast_dsl.py                test00fritzfast_smart_home_temperature.py
test00fritzfast_ecostat.py            test00fritzfast_traffic.py
test00fritzfast_energy.py             test00fritzfast_wifi_load.py
```

```
netcat 192.168.munin.master  4949
# munin node at nas-525a.local
list
FritzboxConfig FritzboxConfig.py cpu cpuspeed df df_inode entropy forks fw_packets 
http_loadtime if_enp0s25 if_enp4s0 if_err_enp0s25 if_err_enp4s0 interrupts irqstats 
load memory munin_stats nfsd nfsd4 open_files open_inodes postfix_mailqueue 
postfix_mailvolume proc_pri processes smart_sdc smart_sdd swap 
test00fritzfast_connection_uptime.py test00fritzfast_smart_home_temperature.py test00fritzfast_traffic.py 
threads users vmstat
```

```
service munin-node restart 
tail -f /var/log/munin/munin-node.log

# fritzbox-plugins # fritzfast-plugins
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36368" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36406" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36356" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36394" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36416" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36408" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36420" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:36434" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:26:16 [855778] Service 'test42fritzfast_wifi_load.py' timed out.
2023/01/27-14:26:19 CONNECT TCP Peer: "[::ffff:127.0.0.1]:59700" Local: "[::ffff:127.0.0.1]:4949"
2023/01/27-14:26:36 [855778] Error output from FritzboxConfig:
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 1: import: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 3: class: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 4: the server address of the Fritzbox (ip or name): not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 5: server: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 6: the port the Fritzbox webserver runs on: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 7: port: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 8: the user name to log into the Fritzbox webinterface: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 9: user: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 10: the password to log into the Fritzbox webinterface: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 11: password: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 12: useTls: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 13: Syntax error: "(" unexpected
2023/01/27-14:26:36 [855778] Service 'FritzboxConfig' exited with status 2/0.
2023/01/27-14:26:36 [855778] Error output from FritzboxConfig:
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 1: import: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 3: class: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 4: the server address of the Fritzbox (ip or name): not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 5: server: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 6: the port the Fritzbox webserver runs on: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 7: port: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 8: the user name to log into the Fritzbox webinterface: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 9: user: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 10: the password to log into the Fritzbox webinterface: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 11: password: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 12: useTls: not found
2023/01/27-14:26:36 [855778]    /etc/munin/plugins/FritzboxConfig: 13: Syntax error: "(" unexpected
2023/01/27-14:26:36 [855778] Service 'FritzboxConfig' exited with status 2/0.
2023/01/27-14:26:52 [855778] Error output from test00fritzfast_energy.py:
2023/01/27-14:26:52 [855778]    Couldn't retrieve fritzbox energy stats: 'actPerc'
2023/01/27-14:26:52 [855778] Service 'test00fritzfast_energy.py' exited with status 1/0.
```

```
munin-run  test00fritzfast_ecostat.py
multigraph cpuload
load.value 3
multigraph cputemp
temp.value 51
multigraph ramusage
strict.value 28.1
cache.value 46.2
free.value 25.7

munin-run  test00fritzfast_energy.py
multigraph power
system.value 40
cpu.value 84
wifi.value 38
dsl.value 100
ab.value 0
Couldn't retrieve fritzbox energy stats: 'actPerc'
Warning: the execution of 'munin-run' via 'systemd-run' returned an error. 
This may either be caused by a problem with the plugin to be executed or a failure 
of the 'systemd-run' wrapper. Details of the latter can be found via 'journalctl'.

munin-run  test00fritzfast_dsl.py
multigraph dsl_capacity
recv.value 72908
send.value 34743
multigraph dsl_snr
recv.value 11
send.value 21
multigraph dsl_damping
recv.value 13
send.value 13
multigraph dsl_errors
es_recv.value 8
es_send.value 0
ses_recv.value 0
ses_send.value 0
multigraph dsl_crc
recv.value 0
send.value 0

munin-run  test00fritzfast_link_saturation.py 
multigraph saturation_up
up_realtime.value 332
up_high.value 989
up_default.value 54357
up_low.value 0
maxup.value 12347000
multigraph saturation_down
dn_internet.value 161201
dn_iptv.value 0
maxdown.value 61738000

munin-run  test00fritzfast_wifi_load.py
multigraph bandwidth_24ghz
24ghz_recv.value 12
24ghz_send.value 7
multigraph neighbors_24ghz
24ghz_samechan.value 9
24ghz_otherchans.value 7

munin-run  test00fritzfast_traffic.py
down.value 6814
up.value 5482
```

**Comment by RenWal** [comment]
Woah, that's a lot of things for a single issue. I'll try to go through what I understood:

> Die fritzbox-munin plugins laufen, nicht ganz problemlos mit 06.8x

Connecting to FRITZ!OS < 07.20 is unsupported, as stated in the requirements. I do not have plans to add support for this myself, since I don't have a test device with that firmware. Pull requests are always welcome though!

> wifi_modes.py kopiert wird neighbors, ist das ein Schreibfehler? neighbours

Not sure if I understand this sentence. There is no file `wifi_modes.py` in this repository. Do you mean `fritzbox_wifi_load.py`? Then again, no, "neighbors" is not an error.

> fritzfast_traffic.py kann es sein, dass down und up vertauscht sind? https://github.com/Tafkas/fritzbox-munin/commit/d7929f15fc15d7e132288b262fc025d12a5f498a

This commit is entirely unrelated to fritzbox-munin-fast. It fixes a bug introduced in https://github.com/Tafkas/fritzbox-munin/commit/2dabb02c2c1b8c7c97c71b80c58a7f527b88a116, which was never introduced into this fork. That bug was also not causing up and downlink traffic to be swapped.

To clarify (following the FritzBox manufacturer's specification): Uplink and downlink traffic is viewed from the router to the WAN port. Unless something has changed on AVM's firmware very recently, this should be working. If you're sure it's swapped for you, please provide the exact model name, firmware version (you're mentioning several firmwares in your issue, so I'm not sure which one is affected), and preferably a screenshot of the graph.

>     # munin node at nas-525a.local
>     list
>     FritzboxConfig FritzboxConfig.py

It looks like you're attempting to execute all of the helper files as Munin plugins. Please do not do so. I'm not sure if I understand how you installed the plugins, since you seem to have renamed all of the plugin files. In case you symlinked the files as described in step 6 of the installation instructions, remove the symlink for `FritzboxConfig.py`, then restart `munin-node`. I'll update the documentation to make this more clear.

>     Couldn't retrieve fritzbox energy stats: 'actPerc'

This looks like it *might* be caused by a recent change in the firmware. Do I read your issue correctly in that this occurs on a FritzBox 3490 running firmware 07.30? If so, please do the following:
- sign into your router
- navigate to the energy monitor page
- open your browser's developer tools and select the network monitor tool, so you can see the HTTP requests
- on the energy monitor page, click the refresh button at the bottom right
- this should trigger a request to `data.lua`
- open that request in the network monitor tool of your browser and then post the entire JSON response, so I can check what's up with the `actPerc` field

**Comment by Ldwg2002** [comment]
Dear RenWal,

thanks for digging through my German post and sorry confusing you with unordered details, background info now as table. I am dealing with this github project since three weeks and I am confused by my own. Facing an initial repository (fritzbox-munin/Tafkas) having 5 contributors and 34 forks. Shouldn't it be the other way round? (rhetorical question) Surprisingly I found a more prospective solution for my deployment in your fork (one of the 34, with little less contributors) using multigraph, TR-064, JSON. And the most important thing - I got an answer very soon :-) thanks a lot, so could step forward over the weekend.  

I am interested to build a monitoring solution of two (unusual) FritzBox-deployments (router and access points with manual channel selection and low transmission power of 12 % per antenna), thats it - SOHO scale. A bit IT-skilled, but if I look into your scripts here, I am not a programmer nowadays. 

### the zoo of boxes in my two nets (VPN connected) ###

| | | | |
|-|-|-|-|
| network size      | middle                | small                | munin monitoring with |
| **router**           | 7490os07.29           | 3490os07.30           | fritzbox-fast/RenWal  |
| WiFi 11n 3x3 SU-MIMO | off / in the cellar   | ON / in use           | fritzbox-fast/RenWal  |
| | | | |
| **WiFi-net**         | 8 AccessPoints        | 2 AccessPoints        | my plan to use for  |
| transmission power   | 12 % each             | 12 % each             | these 6.8x type     |
| AP 11n 3x3 SU-MIMO   | 7272os06.88           | 7272os06.88           | fritzbox/Tafkas     |
| AP 11n 2x2 SU-MIMO   | 7390os06.87           | 7390os06.87           | fritzbox/Tafkas     |
| AP 11n 2x2 SU-MIMO   | 3270v3os05.54  7270v3os06.06  |               | not having this |
| AP 11g 1x1           | 3131os04.57           |                       | data at all     |

Yet I am running (fritzbox-munin/Tafkas) on routers (ok, except energy on 07.30) and not working with AP's 6.8x, issued at https://github.com/Tafkas/fritzbox-munin/issues/61, does not concern here. 
Now starting parallel with (fritzbox-munin-fast/RenWal) on the routers (both >07.2x). That is the reason for renaming all your plugins to **fritzfast** to distinguish from **fritzbox**, because some have the same name (may be they are the same). This way it should be possible to use both repos parallel. Do you have any caveats? 

### your support ###

What I addressed in ```fritzbox_wifi_load.py```, thought a misspelling in the option, but it's both neighbor/**American** and neighbour/**UK**, I now reflected myself :-) May be you allow both, to avoid any trouble. 

```uplink and downlink traffic``` thanks for the clarification.

```I'll update the documentation to make this more clear.``` Thanks a lot for that and sorry for my fault. 
The documentation was my main obstacle as newbie in the whole projects here. I did the same way as worked in fritzbox-munin/Tafkas (unclear too) and made sym.links to FritzboxConfig.py but two times, one as FritzboxConfig.py and the second one FritzboxConfig only. So I did **Remove the symlink for FritzboxConfig.py, then restart munin-node.** and still have the same  
```Error 1 - 13: import: not found``` now only once.  

```Couldn't retrieve fritzbox energy stats: 'actPerc'```

Puh, your task to dig into the depth of http-analyzing was a challenge for us both, describing for you and doing for me. But I am quite sure, that I found what we are looking for 3490os07.30: 

> {"pid":"energy","hide":{"mobile":true,"liveTv":true,"provServ":true,"shareUsb":true,"ssoSet":true},"time":[],"data":{"drain":[{"cumPerc":37,"actPerc":36,"name":"FRITZ!Box Gesamtsystem","statuses":"aktiv seit 4 Tagen, 11 Stunden, 10 Minuten"},{"cumPerc":72,"actPerc":67,"name":"FRITZ!Box Hauptprozessor","statuses":""},{"cumPerc":38,"actPerc":38,"name":"WLAN","statuses":["WLAN aktiviert","2 WLAN-Netzwerkgeräte angemeldet"]},{"cumPerc":100,"actPerc":100,"name":"DSL","statuses":["Internetverbindung hergestellt"]},{"cumPerc":0,"actPerc":0,"name":"USB-Geräte","statuses":"kein USB-Gerät angeschlossen"},{"statuses":"4 Geräte angeschlossen","name":"LAN-Anschlüsse","lan":[{"name":"LAN 1","class":"green"},{"name":"LAN 2","class":"green"},{"name":"LAN 3","class":"green"},{"name":"LAN 4","class":"green"}]}],"intro":{"text":["Auf dieser Seite finden Sie Informationen zum Energieverbrauch der FRITZ!Box und zu belegten LAN-Schnittstellen."]}},"sid":"3a07118d0037bd4c"}

### my current errors ###

```
test00fritzfast ERRORS with both router configs  3490os07.30  and  7490os07.29

2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 1: import: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 3: class: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 4: the server address of the Fritzbox (ip or name): not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 5: server: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 6: the port the Fritzbox webserver runs on: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 7: port: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 8: the user name to log into the Fritzbox webinterface: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 9: user: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 10: the password to log into the Fritzbox webinterface: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 11: password: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 12: useTls: not found
2023/01/29-17:15:24 [787818]    /etc/munin/plugins/FritzboxConfig: 13: Syntax error: "(" unexpected
2023/01/29-17:15:24 [787818] Service 'FritzboxConfig' exited with status 2/0.

test00fritzfast pointing to 3490os07.30 

2023/01/29-17:15:38 [787818] Error output from test00fritzfast_energy.py:
2023/01/29-17:15:38 [787818]    Couldn't retrieve fritzbox energy stats: 'actPerc'
2023/01/29-17:15:38 [787818] Service 'test00fritzfast_energy.py' exited with status 1/0.

test00fritzfast pointing to 7490os07.29
no errors! except my fault # on this router wifi is OFF, see table above

2023/01/29-17:31:28 [803203] Error output from test00fritzfast_wifi_load.py:
2023/01/29-17:31:28 [803203]    Couldn't retrieve fritzbox wifi load: '24ghz'
2023/01/29-17:31:28 [803203] Service 'test00fritzfast_wifi_load.py' exited with status 1/0.
```

```FritzboxConfig``` any idea what causes these messages? Or some commands for investigation? Your fast-plugins seem to work as expected on both boxes, as shown at the end of my inintial post. Energy has low importance for me. 

Thanks a lot for your support. Yours Ludwig

**Comment by Ldwg2002** [comment]
![2023_fritzbox_EnergMon](https://user-images.githubusercontent.com/122236789/215344825-f1d3102a-1419-432e-a42a-212fc101ff1d.jpg)

**Comment by RenWal** [comment]
>  I did the same way as worked in fritzbox-munin/Tafkas (unclear too) and made sym.links to FritzboxConfig.py but two times, one as FritzboxConfig.py and the second one FritzboxConfig only. So I did Remove the symlink for FritzboxConfig.py, then restart munin-node. and still have the same
Error 1 - 13: import: not found now only once.

Please also remove the `FritzboxConfig` symlink.

> `Couldn't retrieve fritzbox energy stats: 'actPerc'`

Huh, this one is interesting. Am I right to assume that you have the answering machine disabled in that FritzBox? I think doing so hides its energy statistics completely on 7.30, which then causes `fritzbox_energy.py` to run over the end of the energy statistics array.

Before I make a permanent change that fixes this, can you please try the following?

- Edit your local copy of `fritzbox_energy.py` in [line 31](https://github.com/RenWal/fritzbox-munin-fast/blob/master/src/fritzbox_energy.py#L31) and remove `'ab', ` from the `DEVICES` array
- Restart `munin-node`
- See if the error has disappeared

**Comment by Ldwg2002** [comment]
So it is good to know a bit about the circumstances, to make the right assumptions. You are right. In the menue phone I don't use anything. 10 phone numbers are listed but not used for anything. In mobile phone times nobody is missing. Later we have to consider, if a newer OS>07.20 is used as AccessPoint only the "dsl" part is OFF too. Is it removed?. And "wifi"?

```
fritzbox_energy.py line 31
DEVICES = ['system', 'cpu', 'wifi', 'dsl', 'usb', 'lan']
```
test00fritzfast pointing on 7490os07.29
#######################################
2023/01/29-20:20:22 Munin::Node::Server (type Net::Server::Fork) starting! pid(956998)
Resolved [*]:4949 to [::]:4949, IPv6
Not including resolved host [0.0.0.0] IPv4 because it will be handled by [::] IPv6
Binding to TCP port 4949 on host :: with IPv6
Setting gid to "0 0"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:46938" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:46954" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:46970" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:46982" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:46994" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47010" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47016" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47022" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47040" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47030" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:47042" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:25:46 [960136] Error output from test29fritzbox_power_consumption: ### not our stuff ###
2023/01/29-20:25:46 [960136]    Couldn't retrieve fritzbox power consumption
2023/01/29-20:25:46 [960136] Service 'test29fritzbox_power_consumption' exited with status 1/0. ### not our stuff ###
2023/01/29-20:26:09 [960104] Error output from test00fritzfast_wifi_load.py: ### is OFF ###
2023/01/29-20:26:09 [960104]    Couldn't retrieve fritzbox wifi load: '24ghz'
2023/01/29-20:26:09 [960104] Service 'test00fritzfast_wifi_load.py' exited with status 1/0.

test00fritzfast pointing on 3490os07.30 
#######################################
2023/01/29-20:28:33 Server closing!
2023/01/29-20:28:33 Munin::Node::Server (type Net::Server::Fork) starting! pid(962220)
Resolved [*]:4949 to [::]:4949, IPv6
Not including resolved host [0.0.0.0] IPv4 because it will be handled by [::] IPv6
Binding to TCP port 4949 on host :: with IPv6
Setting gid to "0 0"
2023/01/29-20:28:37 [962220] Error output from test00fritzfast_dsl.py:
2023/01/29-20:28:37 [962220]    Traceback (most recent call last):
2023/01/29-20:28:37 [962220]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 150, in <module>
2023/01/29-20:28:37 [962220]        print_config()
2023/01/29-20:28:37 [962220]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 115, in print_config
2023/01/29-20:28:37 [962220]        max = retrieve_max_values()
2023/01/29-20:28:37 [962220]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 107, in retrieve_max_values
2023/01/29-20:28:37 [962220]        jsondata = json.loads(data)[0]
2023/01/29-20:28:37 [962220]      File "/usr/lib/python3.9/json/__init__.py", line 346, in loads
2023/01/29-20:28:37 [962220]        return _default_decoder.decode(s)
2023/01/29-20:28:37 [962220]      File "/usr/lib/python3.9/json/decoder.py", line 337, in decode
2023/01/29-20:28:37 [962220]        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
2023/01/29-20:28:37 [962220]      File "/usr/lib/python3.9/json/decoder.py", line 355, in raw_decode
2023/01/29-20:28:37 [962220]        raise JSONDecodeError("Expecting value", s, err.value) from None
2023/01/29-20:28:37 [962220]    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2023/01/29-20:28:37 [962220] Service 'test00fritzfast_dsl.py' exited with status 1/0.
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50580" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50588" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50604" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50608" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50624" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50632" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50628" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50648" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50654" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50664" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:02 CONNECT TCP Peer: "[::ffff:127.0.0.1]:50678" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:30:08 [963930] Error output from test00fritzfast_dsl.py:
2023/01/29-20:30:08 [963930]    Traceback (most recent call last):
2023/01/29-20:30:08 [963930]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 150, in <module>
2023/01/29-20:30:08 [963930]        print_config()
2023/01/29-20:30:08 [963930]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 115, in print_config
2023/01/29-20:30:08 [963930]        max = retrieve_max_values()
2023/01/29-20:30:08 [963930]      File "/etc/munin/plugins/test00fritzfast_dsl.py", line 107, in retrieve_max_values
2023/01/29-20:30:08 [963930]        jsondata = json.loads(data)[0]
2023/01/29-20:30:08 [963930]      File "/usr/lib/python3.9/json/__init__.py", line 346, in loads
2023/01/29-20:30:08 [963930]        return _default_decoder.decode(s)
2023/01/29-20:30:08 [963930]      File "/usr/lib/python3.9/json/decoder.py", line 337, in decode
2023/01/29-20:30:08 [963930]        obj, end = self.raw_decode(s, idx=_w(s, 0).end())
2023/01/29-20:30:08 [963930]      File "/usr/lib/python3.9/json/decoder.py", line 355, in raw_decode
2023/01/29-20:30:08 [963930]        raise JSONDecodeError("Expecting value", s, err.value) from None
2023/01/29-20:30:08 [963930]    json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2023/01/29-20:30:08 [963930] Service 'test00fritzfast_dsl.py' exited with status 1/0.
2023/01/29-20:30:12 [963930] Error output from test00fritzfast_dsl.py:
2023/01/29-20:30:12 [963930]    Couldn't retrieve fritzbox dsl stats: list index out of range
2023/01/29-20:30:12 [963930] Service 'test00fritzfast_dsl.py' exited with status 1/0.
2023/01/29-20:30:17 [963930] Error output from test00fritzfast_link_saturation.py:
2023/01/29-20:30:17 [963930]    Couldn't retrieve fritzbox link saturation: Expecting value: line 1 column 1 (char 0)
2023/01/29-20:30:17 [963930] Service 'test00fritzfast_link_saturation.py' exited with status 1/0.
2023/01/29-20:30:32 [963930] Error output from test00fritzfast_ecostat.py:
2023/01/29-20:30:32 [963930]    ERROR: Did not receive valid JSON data from FritzBox, so automatically cleared the session, please try again.: Expecting value: line 1 column 1 (char 0)
2023/01/29-20:30:32 [963930] Service 'test00fritzfast_ecostat.py' exited with status 1/0.
2023/01/29-20:30:34 [963949] Error output from 29fritzwifi_memory_usage:
2023/01/29-20:30:34 [963949]    Couldn't retrieve fritzbox memory usage
2023/01/29-20:30:34 [963949] Service '29fritzwifi_memory_usage' exited with status 1/0.
2023/01/29-20:30:35 [963936] Error output from test29fritzbox_wifi_devices:
2023/01/29-20:30:35 [963936]    Couldn't retrieve connected fritzbox wifi devices
2023/01/29-20:30:35 [963936] Service 'test29fritzbox_wifi_devices' exited with status 1/0.
2023/01/29-20:30:53 [963936] Error output from test29fritzbox_power_consumption:
2023/01/29-20:30:53 [963936]    Couldn't retrieve fritzbox power consumption
2023/01/29-20:30:53 [963936] Service 'test29fritzbox_power_consumption' exited with status 1/0.
2023/01/29-20:31:02 [963930] Error output from test00fritzfast_energy.py:
2023/01/29-20:31:02 [963930]    Couldn't retrieve fritzbox energy stats: 'actPerc'
2023/01/29-20:31:02 [963930] Service 'test00fritzfast_energy.py' exited with status 1/0.

**Comment by Ldwg2002** [comment]
FritzboxConfig symlink is deleted - no errors

**Comment by Ldwg2002** [comment]
I was too fast, with munin it takes some 5 to 10 minutes. This is the only error ```fritzfast``` concerning
2023/01/29-20:50:01 CONNECT TCP Peer: "[::ffff:127.0.0.1]:39752" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:50:01 CONNECT TCP Peer: "[::ffff:127.0.0.1]:39766" Local: "[::ffff:127.0.0.1]:4949"
2023/01/29-20:50:14 [985628] Error output from test00fritzfast_energy.py:
2023/01/29-20:50:14 [985628]    Couldn't retrieve fritzbox energy stats: 'actPerc'
2023/01/29-20:50:14 [985628] Service 'test00fritzfast_energy.py' exited with status 1/0.

```
munin-run  test00fritzfast_energy.py
multigraph power
system.value 38
cpu.value 77
wifi.value 38
dsl.value 100
ab.value 0
Couldn't retrieve fritzbox energy stats: 'actPerc'
Warning: the execution of 'munin-run' via 'systemd-run' returned an error. 
This may either be caused by a problem with the plugin to be executed or 
a failure of the 'systemd-run' wrapper. Details of the latter can be found via 'journalctl'.
```

**Comment by Ldwg2002** [comment]
sorry, just double checked the change in fritzbox_energy.py line 31 was not saved.

**Comment by Ldwg2002** [comment]
I can report full success to you. **No errors concerning fritzfast.** Now I have to fiddle with the right munin config to show my first multigraphs of test00fritzfast.

FritzboxConfig symlink is deleted - no errors

So it is good to know a bit about the circumstances, to make the right assumptions. You are right. In the menue phone I don't use anything. 10 phone numbers are listed but not active. In mobile phone times nobody is missing. Later we have to consider, if a newer OS>07.20 is used as AccessPoint only the "dsl" part is OFF too. Is it removed?. And "wifi"?

```
fritzbox_energy.py line 31
DEVICES = ['system', 'cpu', 'wifi', 'dsl', 'usb', 'lan']
```

```
munin-run  test00fritzfast_energy.py
multigraph power
system.value 38
cpu.value 76
wifi.value 38
dsl.value 100
usb.value 0
multigraph devices
wifi.value 3
lan.value 4
multigraph uptime
uptime.value 5.03
```
Thanks a lot. Yesterday I tried my **7490** spare box with Labor**OS 07.51** because I intend tu use WireGuard. It worked without problems, ecen AVM altered the design a bit and the webGUI feels a little faster. I will continue and keep the 7490os07.29 as spare part.

**Comment by RenWal** [comment]
Support questions addressed. Closing in favor of #6 and #7

**Comment by Ldwg2002** [comment]
thank's for all.
